src/MAGMA.jl

```diff
@@ -23,15 +23,140 @@ const MagmaSomeVec       = 305
 const MagmaOverwriteVec  = 306
 const MagmaBacktransVec  = 307
 
+# https://github.com/mweastwood/MAGMA.jl
+
+export device_reset,
+       MagmaVector, MagmaMatrix, get,
+       axpy!, dot,
+       gemv!
+
+import Base: length,size,dot
+
+
+# some types' definitions and their accordingly chararcters
+const magmatypes = Union{Float32, Float64, ComplexF32, ComplexF64}
+const magmatypeslist =  (Float32,Float64,ComplexF32,ComplexF64)
+const typechar = Dict(
+    Float32 => 's',
+    Float64 => 'd',
+    ComplexF32 => 'c',
+    ComplexF64 => 'z'
+)
+
 """
 the path to magma binary
 """
 const libmagma = "/usr/local/magma/lib/libmagma.so"
 
+
+# https://github.com/mweastwood/MAGMA.jl
+struct MagmaVector{T <: magmatypes}
+    ptr::Ptr{T}
+    length::Int
+end
+
+struct MagmaMatrix{T <: magmatypes}
+    ptr::Ptr{T}
+    size::Tuple{Int, Int}
+end
+
+length(x::MagmaVector) = x.length
```

Comment:
overload the `Base.length`, please learn how to overload generic functions from docs